spec.bs

```diff
@@ -18,3 +18,50 @@ Default Biblio Status: current
 Markup Shorthands: markdown yes
 Assume Explicit For: yes
 </pre>
+
+
+<h2 id="advertisements">Advertisements</h2>
+
+An <dfn export>advertisement</dfn>
+is a [=struct=] with the following [=struct/items=]:
+
+* An <dfn for=advertisement>render URL</dfn>. The render URL must have “https” as the scheme.
+* An optional <dfn for=advertisement>metadata</dfn> which is text containing opaque JSON encoded data, passed as an object to auction worklet.
+
+
+<h2 id="advertisement-interest-groups">Advertisement Interest Groups</h2>
+
+An <dfn export>advertisement interest group</dfn>
+is a [=struct=] with the following [=struct/items=]:
+ 
+* An <dfn for="advertisement interest group">expiration</dfn> date.   base::Time expiry;
+
+* An <dfn for="advertisement interest group">owner origin</dfn>, an [=origin=] url.

```

Comment:
Well, just "origin" wouldn't read well to me since I wouldn't know it is a URL without mentioning it somewhere.  It might be another kind of object that stores info about the owner origin, for example.  The name could be "owner origin URL", similar to "render URL".   I don't know if there is any style guideline about this.

Reply:
Since the origin value is actually a string, intended to be parsed to an origin, I changed the text to: "a [=valid url string=] whose [=url/origin=] will be used as the owner of the interest group.", at Jeffrey's suggestion.

Reply:
It'll be different between the WebIDL dictionary that's an argument to `joinAdInterestGroup`, and the internal storage of the interest groups the browser has joined. +1 that the internal storage needs to use just `[=origin=]`, and not say this field is a string or URL. I suggested to @dlaliberte that he start with the public API instead of the internal storage, partly to keep everything grounded in actual algorithms, and partly to avoid the pile of "unexported definition" warnings that come from having a bunch of unused `<dfn>` tags.

Reply:
Changed to "An <dfn export for="AuctionAdInterestGroup">owner</dfn> [=origin=]."